### PR TITLE
Add 2D tests for proton boron fusion

### PR DESCRIPTION
**Pull request by NeilZaim**
This PR is based on #2450, which is based on #2330, so it should probably be reviewed once these PRs have been merged.

It adds the same automated tests as in #2450, but in 2D instead of 3D. The same analysis script is used to analyze both tests, with only minor modifications.

Putting the [WIP] tag until I have tested this on GPU.
Edit: tested on GPU so I'm removing the WIP tag.

### CODE REVIEW

File: Examples/Modules/nuclear_fusion/inputs_2d [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+my_constants.proton4_energy = 550*keV_to_J
+
+proton4.species_type = proton
+proton4.injection_style = "NRandomPerCell"
+proton4.num_particles_per_cell = 800
+proton4.profile = "constant"
+proton4.density = 1.e35
+proton4.momentum_distribution_type = "constant"
+proton4.uz = sqrt(2*m_p*proton4_energy)/(m_p*clight)
+proton4.do_not_push = 1
```
Comment (ax3l) [inline Examples/Modules/nuclear_fusion/inputs_2d:136]: as in 3D: also let us add `do_not_deposit`

File: Examples/Modules/nuclear_fusion/inputs_2d [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+#################################
+###### Boundary Condition #######
+#################################
+boundary.field_lo = periodic periodic
+boundary.field_hi = periodic periodic
+
+#################################
+############ NUMERICS ###########
+#################################
+warpx.serialize_ics = 1
```
Comment (ax3l) [inline Examples/Modules/nuclear_fusion/inputs_2d]: let's move out to `.ini`

File: Regression/WarpX-tests.ini [hunk 3]
```diff
@@ -1788,6 +1788,36 @@ analysisRoutine = Examples/Tests/collision/analysis_collision_2d.py
...
+useOMP = 1
+numthreads = 2
+compileTest = 0
+doVis = 0
+analysisRoutine = Examples/Modules/nuclear_fusion/analysis_proton_boron_fusion.py
+
+[Proton_Boron_Fusion_2D]
+buildDir = .
+inputFile = Examples/Modules/nuclear_fusion/inputs_2d
+runtime_params =
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]:
```suggestion
runtime_params = warpx.do_dynamic_scheduling=0 warpx.serialize_initial_conditions=1
```

**Comment by NeilZaim** [comment]
I think this PR is ready for review @RemiLehe @ax3l, thanks!

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**